pkg/manifests/manifests_test.go

```diff
@@ -211,19 +217,23 @@ func TestManifests(t *testing.T) {
 		t.Errorf("expected liveness probe host to be \"localhost\", got %q", deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.Handler.HTTPGet.Host)
 	}
 
-	if deployment.Spec.Template.Spec.Volumes[0].Secret == nil {
-		t.Error("router Deployment has no secret volume")
+	if nVolumes := len(deployment.Spec.Template.Spec.Volumes); nVolumes != 2 {
+		t.Errorf("expected router Deployment to have 2 volumes, found %d", nVolumes)
 	}
 	if deployment.Spec.Template.Spec.Volumes[0].Secret.SecretName != secretName {
 		t.Errorf("expected router Deployment volume with secret %s, got %s",
 			secretName, deployment.Spec.Template.Spec.Volumes[0].Secret.SecretName)
 	}
+	if deployment.Spec.Template.Spec.Volumes[1].Secret.SecretName != metricsSecretName {
```

Comment:
Refer volume by name and then check the secret?

Reply:
This code was removed per your other comment.